src/ui-kit/experimental/hierarchical/hierarchical-tree-grid/hierarchical-tree-grid.component.ts

```diff
@@ -36,116 +35,149 @@ export interface GridItem {
+
+  /**
+ * Event emitted when level change is clicked
+ */
   @Output() public levelChanged = new EventEmitter<GridItem>();
-  public selectedItem: GridItem;
-  public chkSelected: boolean = false;
-  selectedItemIndex = 0;
 
-  public displayedColumns = ['select'];
-  public samTableDataSource: any | null;
-  dataChange: BehaviorSubject<any[]> = new BehaviorSubject<any[]>([]);
-  public focusedCell: any;
-  public focusedtemIndex = 0;
+  /**
+ * Event emitted when row is clicked
+ */
+  @Output() public rowChanged = new EventEmitter<GridItem>();
 
+  /**
+  * Event emitted when row set is selected.
+  */
+  @Output() selectResults = new EventEmitter<any[]>();
 
-  @ViewChild(SamPaginationComponent) paginator: SamPaginationComponent;
-  @ViewChild(SamSortDirective) sort: SamSortDirective;
-  @ViewChild('filter') filter: ElementRef;
+  public displayedColumns = ['select'];
 
+  public selectedList: any[] = [];
+  public hierarchicalDataSource: HierarchicalDataSource | null;
+  public dataChange: BehaviorSubject<any[]> = new BehaviorSubject<any[]>([]);
+  @ViewChild(SamSortDirective) sort: SamSortDirective;
 
   ngOnChanges() {
-    this.displayedColumns = [...this.displayedColumns, ...this.templateConfigurations.displayedColumns];
+    this.dataChange.next(this.gridData);
+    if (this.hierarchicalDataSource) {
+      this.hierarchicalDataSource.filter = this.filterText;
+    }
   }
 
   ngOnInit() {
-    this.dataChange.next(this.dataSource);
-    this.samTableDataSource = new SampleDataSource(
+    this.displayedColumns = [...this.displayedColumns, ...this.templateConfigurations.displayedColumns];
+  }
+
+  ngAfterViewInit() {
+    this.hierarchicalDataSource = new HierarchicalDataSource(
       this.dataChange,
-      this.paginator,
       this.sort
     );
   }
-
-  onSelectItem(ev: Event, item: GridItem) {
-
-    this.selectedItem = item;
-    this.itemSelected.emit(this.selectedItem);
-
-  }
-  handleKeyup(ev) {
-    if (KeyHelper.is('tab', event)) {
-      return
-    }
-    if (KeyHelper.is('down', event)) {
-      console.log('onDownArrowDown')
+  /**
+   * On select the results
+   */
+  onChecked(ev, row) {
+    if (ev.target.checked) {
+      this.selectedList = [...this.selectedList, row];
+    } else {
+      const index: number = this.selectedList.indexOf(row);
+      if (index !== -1) {
+        this.selectedList = this.selectedList.filter(item => item !== row);
+      }
     }
-
-    // On up arrow press
-    if (KeyHelper.is('up', event)) {
-      console.log('onUpArrowDown')
-    }
-
-    console.log(ev)
+    this.selectResults.emit(this.selectedList);
   }
 
-  isSelected(item: any) {
-    return this.selectedItem ?
-      this.focusedCell.id == item.id : false;
-  }
+  /**
+   * On level change 
+   */
   public onChangeLevel(ev: Event, item: GridItem): void {
     this.levelChanged.emit(item);
 
   }
+  /**
+  * when the row is click updates the table data
+  */
+  onRowChange(ev, row) {
```

Comment:
@yerramshilpa Same here -- can you make sure you add type information for all your methods?